Makefile

```diff
@@ -574,12 +574,21 @@ misspell:
 yamllint:
 	hack/check/check-yaml.sh
 
+
+# These are the modules that still contain issues reported by golangci-lint.
+# Once a module is updated to be lint-free, remove from list to catch lint regressions.
+MODULES_NEEDING_LINT_FIX=./cmd/cli/plugin/login ./cmd/cli/plugin/package ./cmd/cli/plugin-admin/builder . ./apis/core ./apis/addonconfigs ./capabilities/client ./addons ./hack/packages/package-tools ./packageclients ./tkg ./tkr ./pinniped-components/post-deploy ./pkg/v1/tkr
```

Comment:
can we use `skip-dirs` in `.golangci.yaml` instead ?

https://github.com/vmware-tanzu/tanzu-framework/blob/c84836a4c5a1d9c7b5f3f9b244537f9e71d3c888/.golangci.yaml#L155

ref: https://golangci-lint.run/usage/configuration/#run-configuration

Reply:
@vuil : once this is in, we can mob on taking the one dir at a time and fixing the lint issues. WDYT ?

Reply:
My understanding is, we would like to run the linter jobs on all the modules so that the errors are shown as part of the CI logs, however only block the PR for the modules that has all the linter errors fixed.

Using `skip-dirs` will completely skip the job run on those directories.